src/cluster_legacy.c

```diff
@@ -3621,16 +3621,21 @@ int clusterProcessPacket(clusterLink *link) {
 
         /* Check for role switch: replica -> primary or primary -> replica. */
         if (sender) {
-            serverLog(LL_DEBUG, "node %.40s (%s) announces that it is a %s in shard %.40s", sender->name,
-                      sender->human_nodename, sender_claims_to_be_primary ? "primary" : "replica", sender->shard_id);
             if (sender_claims_to_be_primary) {
                 /* Node is a primary. */
-                clusterSetNodeAsPrimary(sender);
+                if (nodeIsReplica(sender)) {
```

Comment:
use `sender_last_reported_as_primary` so it is more symmetric to the other branch?
```suggestion
                if (sender_last_reported_as_replica) {
```